smac/utils/validate.py

```diff
@@ -0,0 +1,383 @@
+        repetitions: int
+            number of seeds per instance/config to be evaluated
+        runhistory: RunHistory or None
+            if given, try to reuse these results and save some runs
+
+        Returns
+        -------
+        runs: list<dict<string,string,string,string>>
+            list with dicts (config/inst/seed/inst_specs), that can be fed into tae
+        """
+        # If no instances are given, fix the instances to one "None" instance
+        if len(insts) == 0:
+            insts = [None]
+        # If algorithm is deterministic, fix repetitions to 1
+        if self.scen.deterministic:
+            self.logger.debug("Fixing repetitions to one, because algorithm is"
+                              " deterministic.")
+            repetitions = 1
+
+        # Extract relevant information from given runhistory
+        inst_seed_config = self._process_runhistory(configs, insts, runhistory)
+
+        # Now create the actual run-list
+        runs = []
+        # Counter for imputed runs
+        imputed = 0
+
+        for i in sorted(insts):
+            for rep in range(repetitions):
+                configs_evaluated = []
+                if runhistory and i in inst_seed_config:
+                    # Choose seed based on most often evaluated inst-seed-pair
+                    seed, configs_evaluated = inst_seed_config[i].pop(0)
+                    # Delete i from dict if list is empty
+                    if len(inst_seed_config[i]) == 0:
+                        inst_seed_config.pop(i)
+                    # Add runs to runhistory
+                    for c in configs_evaluated:
+                        runkey = RunKey(runhistory.config_ids[c], i, seed)
+                        cost, time, status, additional_info = runhistory.data[runkey]
+                        self.rh.add(c, cost, time, status, instance_id=i,
+                                    seed=seed, additional_info=additional_info)
+                        imputed += 1
+                else:
+                    # If no runhistory or no entries for instance, get new seed
+                    seed = self.rng.randint(MAXINT)
+                    if self.scen.deterministic:
+                        seed = 0
+                # configs in inner loop -> same inst-seed-pairs for all configs
+                for config in [c for c in configs if not c in
+                               configs_evaluated]:
+                    specs = self.scen.instance_specific[i] if i else ""
+                    runs.append({'config':config,
+                                 'inst':i,
+                                 'seed':seed,
+                                 'inst_specs': specs})
+
+        self.logger.info("Collected %d runs from %d configurations on %d instances "
+                         "with %d repetitions.", len(runs), len(configs), len(insts),
+                         repetitions)
+        self.logger.info("Imputed %d runs from given runhistory.", imputed)
```

Comment:
I think the word "imputed" could be confusing for scikit-learn users, because imputed has a very different meaning there. Actually, I can't figure out myself what it means here.